app/models/piece.rb

```diff
@@ -1,53 +1,93 @@
 class Piece < ActiveRecord::Base
   belongs_to :game
 
-  # Returns true/false if a piece is obstructed in its movement.
-  # Errors for movements not diagonal, vertical or horizontal
+  def move_to!(x, y)
+  # Immediately return to controller with false if move is deemed invalid
+    return false if !valid_move? x, y
```

Comment:
Not sure what you're asking ultimately. Seemed to make sense to check first if the proposed piece's move is valid before anything. I'm pretty sure the knight is covered.